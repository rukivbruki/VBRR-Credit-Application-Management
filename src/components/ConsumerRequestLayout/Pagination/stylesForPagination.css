@import '../../../../node_modules/element-plus/lib/theme-chalk/index.css';

.el-pagination .btn-next .el-icon,
.el-pagination .btn-prev .el-icon {
  display: contents;
}
