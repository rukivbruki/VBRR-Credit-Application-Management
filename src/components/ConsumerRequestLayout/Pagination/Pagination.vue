<template>
  <div class="block">
    <el-pagination
      background
      layout="prev, pager, next"
      :total="Number(numberOfPages)"
      @click="btnClick"
      @prev-click="prevBtnClick"
      @next-click="nextBtnClick"
    />
  </div>
</template>

<script>
import './stylesForPagination.css';
export default {
  name: 'Pagination',
  props: ['numberOfPages'],
  data() {
    return {
      clickedBtnNumber: 1,
    };
  },
  methods: {
    incrementClickedBtnNumber() {
      this.clickedBtnNumber++;
    },
    decrementClickedBtnNumber() {
      this.clickedBtnNumber--;
    },
    btnClick(event) {
      const isButton = event.target.tagName === 'LI';
      if (!isButton) {
        return;
      }
      this.$store.dispatch(
        'setPageNumber',
        Number(event.target.firstChild.data),
      );
      this.clickedBtnNumber = Number(event.target.firstChild.data);
    },
    nextBtnClick() {
      this.incrementClickedBtnNumber();
      this.$store.dispatch('setPageNumber', this.clickedBtnNumber);
    },
    prevBtnClick() {
      this.decrementClickedBtnNumber();
      this.$store.dispatch('setPageNumber', this.clickedBtnNumber);
    },
  },
};
</script>

<style scoped>
.block {
  text-align: center;
  margin: 15px;
}
</style>
