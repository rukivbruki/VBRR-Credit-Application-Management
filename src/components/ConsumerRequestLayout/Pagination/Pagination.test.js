import { shallowMount } from '@vue/test-utils';
import Pagination from './Pagination';

describe('Pagination.vue render', () => {
  test('renders correctly', () => {
    expect(shallowMount(Pagination).element).toMatchSnapshot();
  });
});
