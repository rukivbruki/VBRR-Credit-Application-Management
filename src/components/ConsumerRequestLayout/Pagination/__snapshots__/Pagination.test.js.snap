// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination.vue render renders correctly 1`] = `
<div
  class="block"
>
  <el-pagination
    background=""
    layout="prev, pager, next"
    total="NaN"
  />
</div>
`;
